gobblin-core/src/main/java/org/apache/gobblin/source/PartitionedFileSourceBase.java

```
@@ -367,7 +384,25 @@ private long getLowWaterMark(Iterable<WorkUnitState> previousStates, String lowW
     return lowWaterMarkValue + getRetriever().getWatermarkIncrementMs();
   }
 
+  /** Returns the low watermark value based on lookback which is equal to current time minus lookback time. */
+  private long getLowWaterMarkFromLookbackTime(String lookBackTime) {
+    try {
+      Duration lookBackDuration = PartitionAwareFileRetrieverUtils.getLookbackTimeDuration(lookBackTime);
+      return new DateTime().minus(lookBackDuration).getMillis();
+    } catch (IOException e) {
+      Throwables.propagate(e);
```

Should we propagate the exception if we want to have a fallback to 0?

Also, looks like Throwables.propagate is deprecated. Let's just log the exception and move ahead here.

---

replaced with `log.warn("Failed to parse lookback time: {} , Returning 0 as low watermark value, ", lookBackTime, e);`